app/models/validations/sales/sale_information_validations.rb

```diff
@@ -3,18 +3,22 @@ def validate_pratical_completion_date_before_saledate(record)
     return if record.saledate.blank? || record.hodate.blank?
 
     unless record.saledate > record.hodate
-      record.errors.add :hodate, "Practical completion or handover date must be before exchange date"
+      record.errors.add :hodate, I18n.t("validations.sale_information.hodate.must_be_before_exdate")
     end
   end
 
   def validate_exchange_date(record)
     return unless record.exdate && record.saledate
 
-    record.errors.add(:exdate, I18n.t("validations.sale_information.exdate.must_be_before_saledate")) if record.exdate > record.saledate
-
-    return if (record.saledate.to_date - record.exdate.to_date).to_i / 365 < 1
+    if record.exdate > record.saledate
+      record.errors.add :exdate, I18n.t("validations.sale_information.exdate.must_be_before_saledate")
+      record.errors.add :saledate, I18n.t("validations.sale_information.saledate.must_be_after_exdate")
+    end
 
-    record.errors.add(:exdate, I18n.t("validations.sale_information.exdate.must_be_less_than_1_year_from_saledate"))
+    if record.saledate - record.exdate > 1.year
```

Comment:
the previous version of thos (on old line 15) rounds to integers, which means if the difference is between 1 and 2 years the validation won't work as it should - I think this way is also easier to read which is nice

Reply:
I think your new line is much nicer anyway for readability. But in what case didn't the old one work? I would expect it to round down the number of years in the division if the number of days is less than 365 so if the time span was under 1 year it would resolve to 0 < 1

that's not really important given your line is nicer anyway.

As a minor point that looks to me it does want changing, it sounds from the description than you want >= rather than >

Reply:
ah good point, yep I changed it to unless... >  rather than if... < at some point but I see that's different, good spot about >=